src/test/librbd/deep_copy/test_mock_ObjectCopyRequest.cc

```diff
@@ -749,6 +749,8 @@ TEST_F(TestMockDeepCopyObjectCopyRequest, WriteSnaps) {
 TEST_F(TestMockDeepCopyObjectCopyRequest, Trim) {
   ASSERT_EQ(0, m_src_image_ctx->operations->metadata_set(
               "conf_rbd_skip_partial_discard", "false"));
+  m_src_image_ctx->skip_partial_discard = false;
```

Comment:
just wondering, why do we need to set it here? I thought `metadata_set` would also apply the change to the current context.

Reply:
I think it's because it doesn't, by itself, force a refresh of the image.

Reply:
Don't you think it is a `metadata_set` bug?